src/constructors/test/createGlobalStyle.test.js

```diff
@@ -0,0 +1,274 @@
+// @flow
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ReactDOMServer from 'react-dom/server';
+
+import { expectCSSMatches, getCSS, resetStyled, stripComments, stripWhitespace } from '../../test/utils'
+import stringifyRules from '../../utils/stringifyRules'
+import css from '../css'
+import _createGlobalStyle from '../createGlobalStyle';
+import ThemeProvider from '../../models/ThemeProvider'
+import ServerStyleSheet from '../../models/ServerStyleSheet'
+import StyleSheetManager from '../../models/StyleSheetManager'
+
+const createGlobalStyle = _createGlobalStyle(stringifyRules, css)
+const styled = resetStyled();
+
+let context;
+
+beforeEach(() => {
+  context = setup()
+})
+
+afterEach(() => {
+  context.cleanup()
+})
+
+describe(`createGlobalStyle`, () => {
+  it(`returns a function`, () => {
+    const Component = createGlobalStyle``
+    expect(typeof Component).toBe('function')
+  });
+
+  it(`injects global <style> when rendered`, () => {
+    const {render} = context
+    const Component = createGlobalStyle`[data-test-inject]{color:red;} `
+    render(<Component/>)
+    expectCSSMatches(`[data-test-inject]{color:red;} `)
+  });
+
+  it(`injects global <style> when rendered to string`, () => {
+    const sheet = new ServerStyleSheet();
+    const Component = createGlobalStyle`[data-test-inject]{color:red;} `
+    const html = context.renderToString(sheet.collectStyles(<Component />))
+
+    const container = document.createElement('div');
+    container.innerHTML = sheet.getStyleTags();
+    const style = container.querySelector('style');
+
+    expect(html).toBe('');
+    expect(stripWhitespace(stripComments(style.textContent))).toBe('[data-test-inject]{ color:red; } ');
+  });
+
+  it(`supports interpolation`, () => {
+    const {cleanup, render} = setup()
+    const Component = createGlobalStyle`div {color:${props => props.color};} `
+    render(
+      <ThemeProvider theme={{ color: 'orange' }}>
```

Comment:
I don't think the `ThemeProvider` is relevant to this test and makes it confusing.